scalameter-core/src/main/scala/org/scalameter/picklers/primitives.scala

```diff
@@ -0,0 +1,83 @@
+package org.scalameter.picklers
+
+import java.nio.ByteBuffer
+
+
+abstract class PrimitivePickler[T] extends Pickler[T] {
+  protected def bits: Int
+
+  protected def unwrap(from: ByteBuffer): T
+
+  final protected def byteBuffer: ByteBuffer = ByteBuffer.allocate(bytes)
+
+  final def bytes: Int = bits / java.lang.Byte.SIZE
+
+  final def tryUnpickle(a: Array[Byte]): (T, Array[Byte]) = {
+    val (elem, rest) = a.splitAt(bytes)
+    (unwrap(ByteBuffer.wrap(elem)), rest)
+  }
+}
+
+object BytePickler extends PrimitivePickler[Byte] {
+  protected def bits: Int = java.lang.Byte.SIZE
+
+  protected def unwrap(from: ByteBuffer) = from.get(0)
+
+  def pickle(x: Byte): Array[Byte] = byteBuffer.put(x).array()
+}
+
+object BooleanPickler extends PrimitivePickler[Boolean] {
+  protected def bits: Int = java.lang.Byte.SIZE
+
+  protected def unwrap(from: ByteBuffer) = from.get(0) == 1.toByte
```

Comment:
I would be explicit about `0` and `1` for booleans here, since there is a higher chance of detecting corrupted stream.